examples/gno.land/p/demo/blog/util.gno

```diff
@@ -1,7 +1,57 @@
 package blog
 
-import "strings"
+import (
+	"sort"
+	"strings"
+)
 
 func breadcrumb(parts []string) string {
 	return "# " + strings.Join(parts, " / ") + "\n\n"
 }
+
+func (p postSlice) GetPostBySlug(slug string) *Post {
+	for _, post := range p {
+		if post.Slug == slug {
```

Comment:
The goal of an AVL tree is to optimize the lazy loading of necessary data and skip unnecessary posts. Unlike this implementation, an AVL tree is scalable and will efficiently handle larger data sets.